lib/utils/reactivation_authenticator.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:async';
+
+import 'package:chopper/chopper.dart';
+import 'package:coffeecard/cubits/authentication/authentication_cubit.dart';
+import 'package:coffeecard/data/repositories/v1/account_repository.dart';
+import 'package:coffeecard/data/storage/secure_storage.dart';
+import 'package:coffeecard/service_locator.dart';
+
+class ReactivationAuthenticator extends Authenticator {
+  final SecureStorage secureStorage;
+  DateTime? tokenRefreshedAt;
+  final Duration debounce = const Duration(seconds: 10);
+  int _retryCount = 0;
+  final int _retryLimit = 1;
+
+  ReactivationAuthenticator(this.secureStorage);
+
+  bool _canRefreshToken() {
+    if (tokenRefreshedAt == null) {
+      return true;
+    } else {
+      return tokenRefreshedAt!.difference(DateTime.now()) < debounce;
+    }
+  }
+
+  Future<void> _evict() async {
+    final authenticationCubit = sl.get<AuthenticationCubit>();
```

Comment:
The service locator should be moved to the repository, else the code is non-testable